### PR TITLE
feat: simplify & reduce file size

### PR DESCRIPTION
**Pull request by CrabNejonas**
This change simplifies the build process by removing dependencies on `wasm-bindgen`, `wasm-pack` and `wee-alloc`, instead manually passing strings to Wasm memory. This also reduces file size by 2kB.

This also removes panicking, since the places we were panicking before are guaranteed to be safe due to being generated in the build script.

### Reasoning

`wasm-bindgen` adds high level bindings between Rust and JS, but its generated code is more generic than what we need, specifically it forces us to have a Rust allocator to reserve space for the passed strings. We know this is not necessary since there can only ever be one string at a time and strings can safely be overwritten after returning from wasm. This PR therefore forgoes the high level wrappers of `wasm-bindgen` and opts to pass strings using pointer and length pairs. 

Likewise, `wasm-pack` is a nice high level util tool for optimizing and packaging a Wasm&JS lib, but it forces us into weird workaround if we want to have custom JS wrapper code. It also doesn't do much for us since the optimized default build from `rustc` archives the same performance benefits.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,23 +1,25 @@
 {
   "name": "file-icons",
-  "private": true,
   "version": "0.1.0",
-  "description": "Solid CSR Template",
-  "license": "PolyForm-Noncommercial-1.0.0",
+  "description": "File Icons for IDEs",
+  "license": "MIT OR Apache-2.0",
   "author": "CrabNebula <hello@crabnebula.dev>",
```
Comment (lucasfernog-crabnebula) [inline package.json:6]: the author could be better I guess 😂

**Review by lucasfernog-crabnebula — APPROVED** [review]
Rust code could use some cargo-fmt but it looks good.

**Comment by CrabNejonas** [comment]
> Rust code could use some cargo-fmt but it looks good.

believe it or not, it passes rustfmt without changes 🤷

File: package.json (hunk 1 above)
Comment (CrabNejonas) [reply package.json]: Updated it 👍

**Pull request merged**